### PR TITLE
storage: initial commit of objects

### PR DESCRIPTION
**Pull request by rakyll**

### CODE REVIEW

**Comment by rakyll** [comment]
/cc @broady @jba 

Rewrote the sample with your feedback. Closing the other PR.

File: storage/objects/main.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	if len(os.Args) < 2 {
+		usage("missing subcommand")
+	}
+
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
```
Comment (jba) [inline storage/objects/main.go]: I would make the BucketHandle and ObjectHandle here and pass them around. That's the way we should be teaching people to use the storage API. That way the read function, for example, can just take one arg.

File: storage/objects/main.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	if _, err := src.CopyTo(ctx, dst, nil); err != nil {
+		return err
+	}
+	// [END copy_file]
+	return nil
+}
+
+func delete(client *storage.Client, bucket, object string) error {
+	ctx := context.Background()
+	// [START download_file]
```
Comment (jba) [inline storage/objects/main.go]: download_file?

File: storage/objects/main.go (hunk 1 above)
Comment (rakyll) [inline storage/objects/main.go]: The tutorials prepared for these snippets are very independent. Without giving enough context about how BucketHandle and ObjectHandle are constructed, it is very hard for the developers to follow. That's why every snippet is beginning at the client level.

**Comment by rakyll** [comment]
@jba, I initially opened https://github.com/GoogleCloudPlatform/golang-samples/pull/126 that was merging a similar program but decided to rewrite it and abandoned the previous PR.

**Pull request merged**